src/lib.rs

```diff
@@ -390,6 +407,9 @@ pub trait Terminal: Write {
     /// Returns `Ok(true)` if the deletion code was printed, or `Err(e)` if there was an error.
     fn carriage_return(&mut self) -> Result<()>;
 
+    /// Gets the size of the terminal window, if available
+    fn dims(&self) -> Result<Dims>;
```

Comment:
Again, I'd call this either `dimensions` or `size`. `dim` means something else so `dims` is a bit confusing.